### DISCUSSION TITLE
Getting Elevation Data on 2D map

### DISCUSSION THREAD

**Discussion by goksungurhan5** [maplibre/maplibre-gl-js#5113] [Q&A]
I have a pmtiles file that I serve to use 3D terrain maps offline. When I call the `setTerrain` function with this pmtiles file, I can successfully load the 3D terrain and get elevation data using `map.terrain.getElevationForLngLatZoom(maplibregl.LngLat.convert([e.lngLat.lng, e.lngLat.lat]), map.transform.tileZoom)`. **But I want to get this elevation data on the 2D map without calling setTerrain because of performance issues. Is there an easy way I can do this?**

**Comment by HarelM** [comment]
See the following discussion:
- https://github.com/maplibre/maplibre-gl-js/discussions/4721
